include/kf/EarlyAllocator.h

```diff
@@ -24,43 +24,51 @@ namespace kf
 
         template<class Y>
         constexpr EarlyAllocator(const EarlyAllocator<Y>& other) noexcept : m_ptr(reinterpret_cast<T*>(other.m_ptr)), m_size(other.m_size)
-        {
-        }
+        { }
 
-        _CONSTEXPR20_DYNALLOC ~EarlyAllocator() = default;
-        _CONSTEXPR20_DYNALLOC EarlyAllocator& operator=(const EarlyAllocator&) = default;
+        ~EarlyAllocator() = default;
+        EarlyAllocator& operator=(const EarlyAllocator&) = default;
 
         template<POOL_TYPE poolType>
-        T* initialize(const size_t count)
+        T* initialize(const size_t count) noexcept
         {
-            assert(!m_ptr);
-            assert(!m_size);
+            if (m_ptr != nullptr || m_size != 0)
```

Comment:
Let's omit comparing with `nullptr` and `0` and use bool conversion instead:
```
if (m_ptr || m_size)
```
It's easier to read and understand. It's like plain English: "If we have ptr or size then ..."

Reply:
Subjectively I'm not used to comparing all types with bool implicitly, but if this is the style of the project, then ok.